### DISCUSSION TITLE
串列(list)不知如何使用sort()

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#157] [Q&A] [answered]
因為老師在module(就是「模組」)9—串列list的講義中，提到串列相關方法有list.sort()
<img width="960" alt="list sort()問題_1" src="https://user-images.githubusercontent.com/86505454/141972151-47775f89-9362-4ae1-839e-8377188d2989.png">
但，我自己實驗時，都無法成功。
# 使用anaconda中的spyder
<img width="960" alt="list sort()實驗(spyder)_1" src="https://user-images.githubusercontent.com/86505454/141974765-18cfeede-4bb5-469a-b22e-f2d3f8109cd9.png">

# 使用Visual Studio Code(就是VS Code)
<img width="960" alt="list sort()實驗(VS code)_1" src="https://user-images.githubusercontent.com/86505454/141974853-cb54e800-5c85-4117-9125-6e43c3f82550.png">

# 使用jupyter lab
<img width="960" alt="list sort()實驗(jupyter lab)_1" src="https://user-images.githubusercontent.com/86505454/141974903-31c2cd03-0048-488c-909d-b3a3a3306738.png">
不知道問題出在哪。

**Comment by charlottechuang** [comment, marked as answer]
![localhost_8888_lab_tree_%E5%B0%8F%E7%B7%B4%E7%BF%92 ipynb](https://user-images.githubusercontent.com/90754440/142093533-ab0982d6-c513-4255-9ca4-f23812896955.png)

這樣就可以了~

**Reply by enoch-lightning** [reply]
謝謝妳。經過思考，我得出一個結論：`[4,3,2,1].sort()`是一個動作，動作無法使用print()將結果輸出至電腦螢幕。這很合理，因為print()的功用，是把各種型態(type，data type)的物件的內容顯示在電腦螢幕上(顯示在電腦螢幕上就是一種輸出)。因此，若是排序(.sort())會直接更改原串列(list)的值(value，values)，便可以在使用過.sort()後，將排序完的結果使用print()輸出到電腦螢幕上。

**Comment by enoch-lightning** [comment]
我也用jupyter lab照你的方式試，果然成功了。
# jupyter lab
![image](https://user-images.githubusercontent.com/86505454/142155940-882a0e95-512b-472b-950a-b6a8d84adfd3.png)

**Comment by enoch-lightning** [comment]
我有再做一點測試，發現就算變數的名字都叫x，建立值(value)不一樣的串列(list)，每一個串列都放在記憶體(RAM)中的不同位置。
![image](https://user-images.githubusercontent.com/86505454/142156781-51ada947-bdb0-40f5-8cfc-b8da0dd88ef4.png)
![image](https://user-images.githubusercontent.com/86505454/142156829-2c444b70-2ff6-444b-a1b1-b3f268a203a8.png)
程式執行的順序，由先到後是[28]、[29]、[30]、[31]、[32]再[33]。
